### PR TITLE
Parse init and can open

### PR DESCRIPTION
**Pull request by J8cob1**
Initial PR

### CODE REVIEW

**Comment by J8cob1** [comment]
Ready for review at any time

**Review by Boneill3 — CHANGES_REQUESTED** [review]
I updated the EDS test spec so that it runs now. If you run those tests, you will find some are failing. I think this is from changing the camel to snake. There is also something in there that is getting handling attributes in the EDS file. Like parameter name. 

Maybe you can refactor camel_to_snake to make it more clear what is actually camel_to_snake and what is parsing the attribute.

**Comment by J8cob1** [comment]
I fixed the function

File: canopen_monitor/parse/eds.py [hunk 1]
```diff
@@ -2,6 +2,79 @@
...
+
+                new_name += '{}_{}'.format(first, second).lower()
+
+        if (not found_submatch):
+            if (span[0] != 0):
+                new_name += '_'
+
+            new_name += substr.lower()
+
+    return new_name"""
```
Comment (Boneill3) [inline canopen_monitor/parse/eds.py:38]: I don't think we need to keep this as a comment. If there are weird edge cases, I think we should document it with unit tests and we can always pull this up from git if needed. Let me know if you think otherwise!

**Review by Boneill3 — CHANGES_REQUESTED** [review]
I like all the additional comments on camel to snake! There's a lot happening there. 

Do you think camel_to_snake still makes sense for a name for it?

The only change I'm suggesting here is that we don't need to keep the old version in the comments.

File: canopen_monitor/parse/eds.py (hunk 1 above)
Comment (J8cob1) [reply canopen_monitor/parse/eds.py:38]: Oh. I probably forgot to take that out. Thank you for pointing that out

Comment (J8cob1) [reply canopen_monitor/parse/eds.py]: "Do you think camel_to_snake still makes sense for a name for it?"

No. I think it's fine
